Core.IntegrationTests/GitBackedTestBase.cs

```diff
@@ -96,16 +97,31 @@ protected static string ExecuteGitCommandWithOutput (string argument)
     psi.RedirectStandardOutput = true;
     psi.RedirectStandardError = true;
 
-    using var command = Process.Start(psi);
-    command.WaitForExit();
-    if (command.ExitCode != 0)
+    var process = new Process
     {
-      var error = command.StandardError.ReadToEnd();
-    }
+        StartInfo = psi
+    };
+
+    var outputBuilder = new StringBuilder();
+    process.OutputDataReceived += (sender, e) => outputBuilder.Append(e.Data);
+
+    var errorBuilder = new StringBuilder();
+    process.ErrorDataReceived += (sender, e) => errorBuilder.Append(e.Data);
+
+    process.Start();
 
-    var output = command.StandardOutput.ReadToEnd();
+    process.BeginErrorReadLine();
+    process.BeginOutputReadLine();
+
+    process.WaitForExit();
+
+    if (process.ExitCode != 0)
+    {
+      var message = $"Git command failed with error:\n'{errorBuilder}'.";
+      throw new Exception(message);
```

Comment:
Should this be an Assert.Fail(...). Otherwise, it would be InvalidOperationException. It depends on the intention of the error.

If it indicates a violated test expection, it's Assert.Fail, otherwise, it's an InvalidOperationException.